libs/io/tests/test_h5.py

```diff
@@ -0,0 +1,116 @@
+#!/usr/bin/env python3
+# coding: utf-8
+import os
+import shutil
+from pathlib import Path
+from typing import Optional
+
+import h5py
+import numpy as np
+import pytest
+
+from bbhnet.io import h5
+
+
+@pytest.fixture(scope="function")
+def write_dir():
+    write_dir = "tmp"
+    os.makedirs(write_dir, exist_ok=True)
+    yield Path(write_dir)
+    shutil.rmtree(write_dir)
+
+
+@pytest.fixture(scope="function")
+def t():
+    t = [
+        1238163456,
+        12345677890,
+        12345677890,
+        12345677890,
+        12345677890,
+        1238167552,
+    ]
+    return t
+
+
+@pytest.fixture(scope="function")
+def y():
+    y = [23, 53.5, 234, 0, 5, 456.345]
+    return y
+
+
+@pytest.fixture(scope="function")
+def datasets():
+    datasets = {
+        "dataset1": np.arange(6),
+        "dataset2": np.arange(10, 16),
+        "dataset3": np.arange(20, 26),
+        "dataset4": np.arange(30, 36),
+    }
+    return datasets
+
+
+@pytest.fixture(scope="function")
+def prefix():
+    prefix = "out"
+    return prefix
+
+
+def check_file_contents(
+    fname, t, prefix, y: Optional["np.ndarray"] = None, **datasets
+):
+    with h5py.File(fname, "r") as f:
+        # check the timestamps array
+        assert (t == f["GPSstart"][:]).all()
+
+        # if there is "out" array, it should be checked
+        if y:
+            assert (y == f[prefix][:]).all()
+            assert len(y) == len(f[prefix])
```

Comment:
This check is already done implicitly by `y == f[prefix][:]`: if these weren't of equal length, `numpy` would raise an error. More important is to check that `f[prefix]` has the same length as `f["GPSstart"]`, since the premise behind these files is that they are a bunch of functions of time that have a value for each point in the time array.

It's worth adding a check in `write_timeseries` that makes this condition explicit (raising a `ValueError` if the `len` of either `y` or any of the `datasets` doesn't match the `len(t)`), and then adding a test which uses `with pytest.raises(ValueError)` to ensure that this error gets triggered.

Reply:
Agreed. Suggested changes are done
Added the check in writie_timeseries and the test to check if the exception is raised or not